### ISSUE TITLE
queryView method is not working for string parameters.

### ISSUE THREAD

**Issue by gayancharith** [couchbaselabs/react-native-couchbase-lite#99]
According to the documentation I can filter from a key using queryView method. But in my application I can only filter integer values using queryView method. when I use queryView method for strings, it does not filter values from the documents, it just return all the documents from the relevant type. Following way I have defined my views.

            views: {
              person_view: {
                map: 'function (doc) { if (doc.type === "person") { emit(doc.name, null);} }'
              },
            }

following is by filter method.

      filterDocumentByAttribute(view, key) {
        return new Promise((resolve, reject) => {
          let options = {
            key: key,
            include_docs: true
          };
          this.database.queryView(DESIGN_DOCUMENT_NAME, view, options)
          .then((res) => {
            console.log(res.rows);
            resolve(res.rows);
          })
          .catch((error) => {
            console.log(error);
            reject(error);
          });
        });
      }

when the key parameter is integer the method works perfectly, but when it is a string it just return all documents. I have also tried the 'keys' option, but it also does not works for string keys.

**Comment by npomfret** [comment]
Are your keys stored as integers or strings in your map function?

**Comment by gayancharith** [comment]
@npomfret  keys are stored as integers

**Comment by Prudwell** [comment]
@gayancharith Are you using npm version 0.5.4? I just ran into this problem today and after a silly amount of debugging I noticed that a new version of this package was put on npm 2 days ago; including some updates to how the queryView function works. I was on 0.5.3 prior and after updating I can now successfully query using a string as key. Hope this helps

**Comment by jamesnocentini** [comment]
Sorry about that guys, it looks like we need to keep track of changes in a change log more diligently. It would be good to have your feedback on things you'd like to see improve in the plugin. FYI we're working on this PR https://github.com/couchbaselabs/react-native-couchbase-lite/pull/93 to include an http client which has better coverage of the REST API than the one we've been maintaining so far. It will bump the version to 0.6.0 but we're looking for feedback before merging that.

**Comment by gayancharith** [comment]
@Prudwell yeah it is working perfectly for version 0.5.4. Thank you very much.

**Comment by gayancharith** [comment]
@jamiltz @Prudwell but I'm getting a warning for this new version (0.5.4) saying 
`Error return from CBL at PUT http://localhost:5999/testName Response {type: "default", status: 412, ok: false, statusText: undefined, headers: Headers…}`
should I be concerned about this warning.

**Comment by gayancharith** [comment]
I'm using following method to init and start the couchbase lite listner
```
import {manager, ReactCBLite} from 'react-native-couchbase-lite'

ReactCBLite.init((url) => {
      // instantiate a new database
      var database = new manager(url, 'myapp');
      database.createDatabase()
        .then(() => database.getDocuments())
        .then(res => {
          this.setState({
            dataSource: this.state.dataSource.cloneWithRows(res.rows)
          });
        });
    });
```

**Comment by Prudwell** [comment]
Yep I'm getting that too, also a TON of errors upon resetting content and settings and starting up the app, lots of 404 not founds for gets. It seems that in this latest version there was a change to how the final _makeRequest method works.

Old Way
```
  if (res.status == 401) {
        console.log("cbl request failed auth, attempt: " + attemptNumber, settings, fullUrl, res);
```

New Way

```
      if (res.status >= 400) {
        console.warn("Error return from CBL at", settings.method, fullUrl, res);
      }
```

So the old way only gave a console message for 401 unauthorized where the new one will do a console.warn for every failed request. I think our code was silently making lots of 412s (already exists) and 404s before this update as the app looked for documents before being fully ready or doing a createDatabase without checking if the DB already existed. We're only now getting to see those as they happen. I'll keep looking into why my app is going nuts with the warnings on first startup though. If it looks like I can't hide those errors with a catch() or something  I might change that console.warn to a console.log in my local file for development as I'm not sure I need to see it popping up tons of yellow bars for 404s. 

I'll update as I find out more. Cheers

**Comment by npomfret** [comment]
Sorry, I shouldn't have put that logging in.  We'll remove it.
